File: Nejman.Newtone.Core/Data/QueueAction.cs
using Nejman.Newtone.Core.Media;
using System;
using System.Collections.Generic;
using System.Text;

namespace Nejman.Newtone.Core.Data
{
    public static class QueueAction
    {
        public static void Add(MediaSource track)
        {
            if (CoreGlobal.CurrentPlaylist.Count > 0 && CoreGlobal.Audios.ContainsKey(track.Path))
            {
                if (CoreGlobal.QueuePosition < CoreGlobal.CurrentPlaylistPosition || CoreGlobal.QueuePosition > CoreGlobal.CurrentPlaylist.Count)
                {
                    CoreGlobal.QueuePosition = CoreGlobal.CurrentPlaylistPosition;
                }

                CoreGlobal.CurrentPlaylist.Insert(CoreGlobal.QueuePosition + 1, CoreGlobal.Audios[track.Path]);
                CoreGlobal.QueuePosition++;
            }
        }

        public static void Add(string track)
        {
            Add(CoreGlobal.Audios[track]);
        }

        public static void Add(IList<string> tracks)
        {
            foreach(var track in tracks)
            {
                Add(track);
            }
        }

        public static void Add(IList<MediaSource> tracks)
        {
            foreach (var track in tracks)
            {
                Add(track);
            }
        }
    }
}

File: app/src/main/java/com/nejman/nsec/music_player/core/data/QueueAction.java
package com.nejman.nsec.music_player.core.data;

import com.nejman.nsec.music_player.Global;
import com.nejman.nsec.music_player.core.DataContainer;
import com.nejman.nsec.music_player.media.MediaSource;

import java.util.List;

public class QueueAction {
    public static void add(MediaSource track) {
        if (Global.currentPlaylist.size() == 0 || !DataContainer.getInstance().getMediaSources().exists(track.path)) {
            return;
        }

        if (Global.queuePosition < Global.currentPlaylistPosition || Global.queuePosition > Global.currentPlaylist.size()) {
            Global.queuePosition = Global.currentPlaylistPosition;
        }

        Global.currentPlaylist.add(Global.queuePosition + 1, track);
        Global.queuePosition++;
    }

    public static void add(String track) {
        add(DataContainer.getInstance().getMediaSources().get(track));
    }

    public static void add(List<String> tracks) {
        for (String track : tracks) {
            add(track);
        }
    }
}
